comp/syntheticstestscheduler/common/result.go

```diff
@@ -0,0 +1,172 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025-present Datadog, Inc.
+
+package common
+
+// NetpathSource represents the source host of a network path.
+type NetpathSource struct {
+	Hostname string `json:"hostname"`
+}
+
+// NetpathDestination represents the destination host of a network path.
+type NetpathDestination struct {
+	Hostname           string `json:"hostname"`
+	IPAddress          string `json:"ip_address"`
+	Port               int    `json:"port"`
+	ReverseDNSHostname string `json:"reverse_dns_hostname"`
+}
+
+// NetpathHop represents a single hop in a traceroute path.
+type NetpathHop struct {
+	TTL       int     `json:"ttl"`
+	RTT       float64 `json:"rtt"`
+	IPAddress string  `json:"ip_address"`
+	Hostname  string  `json:"hostname"`
+	Reachable bool    `json:"reachable"`
+}
+
+// TracerouteRun represents the result of one traceroute attempt.
+type TracerouteRun struct {
+	Hops []NetpathHop `json:"hops"`
+}
+
+// TracerouteTest aggregates multiple traceroute runs and hop statistics.
+type TracerouteTest struct {
+	TracerouteRuns []TracerouteRun `json:"traceroute_runs"`
+	HopCountAvg    float64         `json:"hop_count_avg"`
+	HopCountMin    int             `json:"hop_count_min"`
+	HopCountMax    int             `json:"hop_count_max"`
+}
+
+// E2ETest represents end-to-end test results such as latency and jitter.
+type E2ETest struct {
+	PacketLoss float64 `json:"packet_loss"`
+	LatencyAvg float64 `json:"latency_avg"`
+	LatencyMin float64 `json:"latency_min"`
+	LatencyMax float64 `json:"latency_max"`
+	Jitter     float64 `json:"jitter"`
+}
+
+// NetpathResult represents the full result of a network path test.
+type NetpathResult struct {
```

Comment:
For netpath payload, please use `pkg/networkpath/payload/pathevent.go` once this PR is merged: https://github.com/DataDog/datadog-agent/pull/40466